### ISSUE TITLE
Performance 

### ISSUE THREAD

**Issue by justinglock40** [FallingSnow/h265ize#118]
Any idea why the CPU and GPU are both getting blasted? I have hevc_nvenc enabled. All the work should be happening on the GPU?

**Comment by justinglock40** [comment]
<img width="652" alt="screenshot at feb 05 09-32-25" src="https://user-images.githubusercontent.com/23133649/35813021-2787267a-0a58-11e8-858e-e2ba16ecaf1e.png">

**Comment by FallingSnow** [comment]
Can you run h265ize with the `--debug` flag and provide the ffmpeg query?

Firstly, cpu will always have to do some work, it is still moving the video frames in and out of memory as well as writing them to disk. From my tests with a 6700hq, I get about 30% cpu and 100% gpu when encoding with intel quicksync.

Second you might be decoding with the cpu and encoding with the gpu, which is probably what is happening here. Maybe you would be able to do both encoding and decoding on the gpu. But even in that case you would still have to use the cpu to move frames: ram -> gpu (decode) -> ram -> gpu(decode) -> ram -> disk. I believe I've read of ways to do it without copying an extra time to ram but I don't think I've ever done it.

TL;DR Either way the cpu is gonna have to feed the gpu and use some gpu, so don't expect 3% cpu usage.

**Comment by justinglock40** [comment]
What flags should be used to encode/decode on the GPU

**Comment by FallingSnow** [comment]
This is outside the scope of h265ize since h265ize doesn't even support hw encoding but you would want to specify a hw decoder codec for the input and a hw encoder codec for the output.
Ex. `ffmpeg -v:c hevc_nvenc -i video.mkv -c:v hevc_nvenc output_video.mkv`
